### PR TITLE
Display batch error

### LINKED ISSUE #3715: Expenditures: As a user I can display error notification when uploading a file
[Provided by Adrian via Contractors project management platform]

## Description
User has to be able to display error notification when uploading a not supported file.

## Acceptance Criteria

- an option to display error when uploading a invalid file
- an option to display error when uploading a file with too many rows

## Design

- [Figma Link - Error state max rows](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=3807%3A289527)
- [Figma Link - Error state invalid file](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=3807%3A291251)

Group 862

reject)

### PR DESCRIPTION
**Pull request by agnieszkajarosikloj**
## Description

This PR adds displaying form errors when file is invalid.
Displaying the locked state has also been added.

Screenshot 2022-08-18 at 16 01 31
Screenshot 2022-08-18 at 16 03 56

[Figma Link - Error state max rows](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=3807%3A289527)
[Figma Link - Error state invalid file](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=3807%3A291251)

## TODO

- Add editing in locked state, when this [PR](https://github.com/JoinColony/colonyDapp/pull/3609) will be merged

Resolves #3715

### CODE REVIEW

**Review by willm30 — COMMENTED** [review]
@agnieszkajarosikloj The max. rows error message is working for me, but even when I do use the template and the CSV below, I get the error that my file structure is incorrect: 

Recipient | Token | Amount
0xb77D57F4959eAfA0339424b83FcFaf9c15407461 | 0x26472Cc7F4cB43ae9b3F18c705De8c34578933F9 | 1000
0xb77D57F4959eAfA0339424b83FcFaf9c15407461 | 0x26472Cc7F4cB43ae9b3F18c705De8c34578933F9 | 1000
0xb77D57F4959eAfA0339424b83FcFaf9c15407461 | 0x26472Cc7F4cB43ae9b3F18c705De8c34578933F9 | 1000
0xb77D57F4959eAfA0339424b83FcFaf9c15407461 | 0x26472Cc7F4cB43ae9b3F18c705De8c34578933F9 | 1000

**Review by Julianahlk — COMMENTED** [review]
The error is showing up for me with a zero:
Screen Shot 2022-08-22 at 10 14 07 AM

**Comment by agnieszkajarosikloj** [comment]
> The error is showing up for me with a zero: Screen Shot 2022-08-22 at 10 14 07 AM

@Julianahlk I couldn't reproduce this error. I changed the condition to show recipients. Could you please check if this error still occurs?

**Comment by Julianahlk** [comment]
The previous code was rendering the recipientsCount when it was set to zero but your recent changes have fixed it. Now this is what I see when I add a form that is empty:

Screen Shot 2022-08-22 at 8 26 29 PM

1. Can we align the error to the left so there is not space in the front?

2. I can't seem to be able to upload any file now, I keep getting this message error:
Screen Shot 2022-08-22 at 8 35 32 PM

Even though, I am using the template I downloaded. My template looks like this:

Empty
Screen Shot 2022-08-22 at 8 38 15 PM

With some values
Screen Shot 2022-08-22 at 8 37 38 PM

**Comment by agnieszkajarosikloj** [comment]
@Julianahlk Thanks for the comment. Token should be added by the token address/id. On [this](https://github.com/JoinColony/colonyDapp/pull/3751) branch you can check which row wasn't imported correctly. 
Please let me know if it helped.

**Comment by agnieszkajarosikloj** [comment]
And I think this file has invalid header. 'template (1)' shouldn't be there, 'Recipient', 'Token' and 'Amount' should be at the top of the table.

**Review by willm30 — APPROVED** [review]
Looking good to me.

**Review by Julianahlk — APPROVED** [review]
> @Julianahlk Thanks for the comment. Token should be added by the token address/id. On [this](https://github.com/JoinColony/colonyDapp/pull/3751) branch you can check which row wasn't imported correctly.

You are correct! I was using my default Apple Numbers spreadsheet application and the format was coming wrong. I tried using Google's spreadsheets and now everything seems to be working well. 👍
Screen Shot 2022-08-24 at 5 14 35 PM

File: src/modules/dashboard/components/ExpenditurePage/Batch/Batch.tsx [hunk 1]
```diff
@@ -68,12 +73,40 @@ interface Props {
...
+    }
+    setValue({
+      ...batch,
+      data: validatedData,
+      recipients: recipientsCount,
+      value: tokens,
+    });
+    setTouched(true, false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [validatedData]);
```
Comment (chinins) [inline src/modules/dashboard/components/ExpenditurePage/Batch/Batch.tsx:97]: Why do you need the hooks dependencies exception here?

**Review by chinins — COMMENTED** [review]
Looking good - nice work. Just a couple of minor comments and a question about useEffect dependencies before I approve.

**Review by chinins — APPROVED** [review]
Looks good!

**Pull request merged**